include/deal.II/multigrid/mg_base.h

```diff
@@ -223,17 +238,39 @@ class MGSmootherBase : public Subscriptor
    * Virtual destructor.
    */
   virtual ~MGSmootherBase();
+
   /**
    * Release matrices.
    */
   virtual void clear() = 0;
 
   /**
-   * Smoothing function. This is the function used in multigrid methods.
+   * Smoothing function that smooths the content in @p u given the right hand
+   * side vector @p rhs. This is the function used in multigrid methods.
    */
   virtual void smooth (const unsigned int level,
                        VectorType         &u,
                        const VectorType   &rhs) const = 0;
+
+  /**
+   * As opposed to the smoothing function, this function applies the action of
+   * the smoothing, overwriting the previous content in the vector u. This
+   * function must be equivalent to the following code
+   * @code
+   * u = 0;
+   * smooth(level, u, rhs);
+   * @endcode
+   * but can usually be implemented more efficiently than the former.
```

Comment:
"If the particular smoother does not implement apply(), the default implementation as described here will be used."